mkdocs_puml/puml.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, base_url: str, num_workers: int = 5, verify_ssl: bool = True)
         self.num_workers = num_workers
         self.verify_ssl = verify_ssl
 
-    def translate(self, diagrams: typing.Iterable[str]) -> typing.List[str]:
+    def translate(self, diagrams: typing.Iterable[str], dark_mode: bool = False) -> typing.List[str]:
```

Comment:
@OnceUponALoop, could you please move `dark_mode` to class attributes? Something like accepting `dark_mode =False` in constructor, then 

```py
self.format = "dsvg" if dark_mode else "svg"
```

this way you'll remove unnecessary conditions

Reply:
Hey man - I gave it a shot but i'm not sure how to make that work (it's friday so it could be my brain being fried.) but if I move it to the class then I have to do the logic in `puml.py` to produce 2 images when dark and one when light.

That would result in different output format for `translate` depending on mode, one of the reasons I voided it initially. I assumed you'd rather keep the puml logic independent from the plugin.

Maybe there's a simpler implementation i'm missing, can you help me out?

```python
def translate(self, diagrams: typing.Iterable[str], dark_mode: bool = False) -> typing.List[str]:
        """Translate string diagram into HTML div
        block containing the received SVG image.

        Examples:
                This method translates content
                into <svg> image of the diagram

        Args:
            diagrams (list): string representation of PUML diagram

        Returns:
            List of SVG images of built diagram (one or two per input diagram if dark_mode is True)
        """
        encoded = [self.preprocess(v) for v in diagrams]

        with ThreadPoolExecutor(max_workers=self.num_workers) as executor:
            if self.dark_mode:
                light_svgs = list(executor.map(lambda ed: self.request(ed, dark_mode=False), encoded))
                dark_svgs = list(executor.map(lambda ed: self.request(ed, dark_mode=True), encoded))
                return list(zip(light_svgs, dark_svgs))
            else:
                svgs = list(executor.map(lambda ed: self.request(ed, dark_mode=False), encoded))
                return svgs

```

Reply:
Hey. I thought about this approach, move `dark mode` attribute to the constructor and build` base_url` right there, for example, (I named this argument as `output_format `)

```py
class PlantUML:
        def __init__(self, base_url: str, num_workers: int = 5, verify_ssl: bool = True, output_format: str = "svg"):
            self.base_url = base_url if base_url.endswith('/') else f"{base_url}/"
            self.base_url = f"{self.base_url}{output_format}/"
```

Then `request` method will remain almost the same

```py
    def request(self, encoded_diagram: str) -> str:
        resp = requests.get(
            urljoin(self.base_url, encoded_diagram),
            verify=self.verify_ssl
        )

        # Use 'ignore' to strip non-utf chars
        return resp.content.decode('utf-8', errors='ignore')
```

The rest of the `PlantUML` is not changed. Then, you can create two  `PlantUML` instances in `plugin.py`

```py
self.puml_light = PlantUML(..., output_format="svg")
self.puml_dark = PlantUML(..., output_format="dsvg")
```

Then use it as

```py
light_svgs = self.puml_light.translate(self.diagrams.values())
dark_svgs = self.puml_dark.translate(self.diagrams.values())
```

This should make the `PlantUML` class a bit more decoupled from the plugin. However, the code still needs x2 time because the dark svgs need to way until light ones are compiled. Ideally, is to map all diagrams under one ThreadPoolExecutor but I suppose it may be delayed to further release.

Reply:
Sounds good, I'll try and get back to this over the weekend :+1:

Reply:
Hi @OnceUponALoop, I plan to prepare a release this weekend. Will you have a chance to look at it? I can edit the PR if you don't mind

Reply:
Sorry for the delay - I implemented your recommendation, modified `on_env`, and updated the associated tests.